pypdf/_writer.py

```diff
@@ -1812,85 +1818,149 @@ def removeLinks(self) -> None:  # deprecated
         deprecation_with_replacement("removeLinks", "remove_links", "3.0.0")
         return self.remove_links()
 
+    def remove_annots(self, subtypes: Optional[Union[str, Iterable[str]]]) -> None:
+        """
+        Remove annotations by Subtype
+        args:
+            subtypes : SubType or list of SubTypes to be removed. None=all
+        """
+        for page in self.pages:
+            self._remove_annots_from_page(page, subtypes)
+
+    def _remove_annots_from_page(
+        self,
+        page: Union[IndirectObject, PageObject, DictionaryObject],
+        subtypes: Optional[Iterable[str]],
+    ) -> None:
+        page = cast(DictionaryObject, page.get_object())
+        if PG.ANNOTS in page:
+            i = 0
+            while i < len(cast(ArrayObject, page[PG.ANNOTS])):
+                an = cast(ArrayObject, page[PG.ANNOTS])[i]
+                obj = cast(DictionaryObject, an.get_object())
+                if subtypes is None or cast(str, obj["/Subtype"]) in subtypes:
+                    if isinstance(an, IndirectObject):
+                        self._objects[an.idnum - 1] = NullObject()  # to reduce PDF size
+                    del page[PG.ANNOTS][i]  # type:ignore
+                else:
+                    i += 1
+
+    def remove_objects_from_page(
+        self,
+        page: Union[PageObject, DictionaryObject],
+        to_delete: Union[ObjectDeletionFlag, Iterable[ObjectDeletionFlag]],
```

Comment:
I think we should make our lives here easier and not allow the Iterables (tuple / list). It is a flag and thus people should use it.

If they don't know it, they will just do multiple calls (which is fine as well)

Reply:
I do not see how to change keeping the code just affecting annotations/comments...